include/solvers/simulation_parameters.h

```diff
@@ -135,6 +135,73 @@ class SimulationParameters
     particlesParameters.parse_parameters(prm);
     void_fraction->parse_parameters(prm);
     multiphysics.parse_parameters(prm);
+
+    // Create output_folder if does not exist
+    //    if (Utilities::MPI::this_mpi_process(this->mpi_communicator) == 0)
+    //      {
+    //        std::filesystem::create_directory(
+    //          simulation_control.output_folder.c_str());
+    //        //        _mkdir(simulation_control.output_folder.c_str());
+    //        std::cout << "Output folder created: "
+    //                  << simulation_control.output_folder << std::endl;
+    //      }
+
+    //    if (simulation_control.make_folder)
+    //      {
+    //        if (mkdir(simulation_control.output_folder.c_str(), 0777) == -1)
+    //          {
+    //            if (errno != EEXIST)
+    //              {
+    //                // if error other than "already exists"
+    //                std::cerr << "Could not create Output folder:  "
+    //                          << strerror(errno) << std::endl;
+    //              }
+    //          }
+    //        else
+    //          std::cout << "Output folder created: "
+    //                    << simulation_control.output_folder << std::endl;
+    //      }
+
+    // Update filenames with the output_folder
+    restart_parameters.filename =
+      simulation_control.output_folder + restart_parameters.filename;
+
+    forces_parameters.force_output_name =
+      simulation_control.output_folder + forces_parameters.force_output_name;
+
+    forces_parameters.torque_output_name =
+      simulation_control.output_folder + forces_parameters.torque_output_name;
+
+    post_processing.kinetic_energy_output_name =
+      simulation_control.output_folder +
+      post_processing.kinetic_energy_output_name;
+
+    post_processing.enstrophy_output_name =
+      simulation_control.output_folder + post_processing.enstrophy_output_name;
+
+    particlesParameters.ib_force_output_file =
+      simulation_control.output_folder +
+      particlesParameters.ib_force_output_file;
+
+    nitsche->force_output_name =
+      simulation_control.output_folder + nitsche->force_output_name;
+
+    nitsche->torque_output_name =
+      simulation_control.output_folder + nitsche->torque_output_name;
```

Comment:
I don't really like this idea of "overwriting" the simulation parameters with additional variable. It's like if we parse something and we overwrite it locally. I think it would be better to use the output_folder manually where we write instead. This would be more coherent with the rest of the library.

Reply:
Do you mean, to ask the user to manually give the "output folder" in multiple subsection? I think it would be very heavy for the user, and a recipe for disaster if you want to launch multiple simulations with sub-directories. 
Or do you mean, we should parse through the code/have a getter method for the `simulation_control.output_folder` so that the `output_name` is directly defined properly in the other parameter objects (`post_processing`, `nitsche` and so on)?

I must say I have great plans for improving folder's management with Lethe ^^ In a later PR, I would like to create automatically a copy of the `.prm` in the `output_folder`, and create a new subfolder in the `output_folder` with the Paraview visualizations. This will really make it easier to check previously launched simulations, and from then we could easily make parametric simulations automatic with an external script (like, you have another script which writes .prm for each parameter, then launches all the simulations one after the other, and the results are neatly stored in their folder). I could tell you more about that ^^

Reply:
What I mean is that I would not merge the output path and the output file name within the parameter itself. I would keep the two parameters seperate and carry them everywhere. I want to avoid "overwriting" a parameter once the parameter file has been read. It gives a very weird use experience (that parameters are being overwritten). I would just carry both the path and the output name and use it where you need it. E.g. When you want to write the force file, the file name would be = param.output_path + param.force_file_name.

Reply:
Alright, I will see to that